### PR TITLE
Add failing unit test

### PR DESCRIPTION
**Pull request by mransan**
Add a unit test which is currently failing for me but I believe should work. (negative int64 of reasonable range should be valid int).

I am running this test on my Mac against the github OCaml. I will run this test at work too on Linux.

### CODE REVIEW

File: pkg/build.ml [hunk 1]
```diff
@@ -6,6 +6,8 @@ let () =
   Pkg.describe "protobuf" ~builder:`OCamlbuild [
     Pkg.lib "pkg/META";
     Pkg.lib ~exts:Exts.module_library "src/protobuf_codec";
+    Pkg.lib ~exts:[".ml"]     "src_test/test_protobuf_codec";
+    Pkg.lib ~exts:[".native"] "src_test/test_protobuf";
```
Comment (whitequark) [inline pkg/build.ml:10]: I don't think you should install tests. `make test` builds and runs them.

Comment (mransan) [inline pkg/build.ml:10]: Somehow `make test` was not working for me. This PR is first to highlight the problem and discuss what should be the solution, I'll clean everything!

Did you manage to reproduce the issue and agree it's in fact not the expected behavior ?

**Comment by whitequark** [comment]
Yeah, I've reproduced (and committed) that, and it does seem like a bug.

**Comment by whitequark** [comment]
On careful thought, none of the intX_of_intY functions in ocaml-protobuf make sense. They should check for the number being within proper range, not just look at 33th/64th bit; they are currently broken for all negative numbers. Patches welcome!

**Comment by mransan** [comment]
I agree, i think it's best to check `Pervasives.min_int` and `Pervasives.max_int`. It's more expensive at runtime but this choice is left to the user of your library so I think it's fine. (They can use `Int32.to_int` and `Int64.to_int` instead). 

Thanks for looking into this!

**Comment by mransan** [comment]
I just checked and your other ppx_deriving_protobuf repo is also using this.

**Pull request closed without merging**